src/main/java/xyz/nucleoid/dungeons/dungeons/item/armor/DgArmorItem.java

```diff
@@ -0,0 +1,148 @@
+    public EquipmentSlot getSlot() {
+        return slot;
+    }
+
+    @Override
+    public double getKnockBackResistance() {
+        return 0;
+    }
+
+    public DgArmorMaterial getMaterial() {
+        return material;
+    }
+
+    public ItemStack createStack(DgItemQuality quality) {
+        return DgArmorUtil.initArmor(DgArmorUtil.armorBuilder(this).build(), quality);
+    }
+
+    @Override
+    public void appendStacks(ItemGroup group, DefaultedList<ItemStack> stacks) {
+        for (DgItemQuality quality : DgItemQuality.values()) {
+            if (group == quality.getItemGroup()) {
+                if (quality.ordinal() >= material.getMinQuality().ordinal() && quality.ordinal() <= material.getMaxQuality().ordinal()) {
+                    stacks.add(createStack(quality));
+                }
+
+            }
+        }
+    }
+
+    @Override
+    public Set<EquipmentSlot> getValidSlots(ItemStack stack) {
+        return ImmutableSet.of(slot);
+    }
+
+    private DgFlavorText generateFlavourText(Random random, DgItemQuality quality, DgArmorMaterial material) {
+        if (quality == null || material == null) {
+            return null;
+        }
+        List<DgFlavorText> choices = new ArrayList<>();
+
+        if (quality.ordinal() <= DgItemQuality.DUSTY.ordinal()) {
+            Collections.addAll(
+                    choices,
+                    flavorTextOf("unimpressive", 1),
+                    flavorTextOf("time", 2),
+                    flavorTextOf("soldier", 3),
+                    flavorTextOf("disease", 2),
+                    flavorTextOf("cookfire", 3)
+            );
+        } else if (quality.ordinal() <= DgItemQuality.FINE.ordinal()) {
+            Collections.addAll(
+                    choices,
+                    flavorTextOf("will_do", 1),
+                    flavorTextOf("unknowledgeable", 3),
+                    flavorTextOf("standard_issue", 3)
+            );
+        } else if (quality.ordinal() <= DgItemQuality.SUPERB.ordinal()) {
+            Collections.addAll(
+                    choices,
+                    flavorTextOf("glow", 2),
+                    flavorTextOf("whispers", 3),
+                    flavorTextOf("craftsmanship", 2),
+                    flavorTextOf("skeleton", 3)
+            );
+        } else {
+            Collections.addAll(
+                    choices,
+                    flavorTextOf("carvings", 2),
+                    flavorTextOf("legendary", 1),
+                    flavorTextOf("reflection", 3),
+                    flavorTextOf("ransom", 2)
+            );
+        }
+        Collections.addAll(
+                choices,
+                flavorTextOf("prized_material", 2),
+                flavorTextOf("expense", 2),
+                flavorTextOf("gram", 3)
+        );
+
+        int idx = random.nextInt(choices.size());
+        return choices.get(idx);
+    }
+
+    @Override
+    public DgFlavorText getFlavorText(Random random, ItemStack stack) {
```

Comment:
Does `DgFlavorTextProvider` assume that this method is stable? I can't remember 😅

Reply:
Alright